.github/workflows/windows-ci.yaml

```diff
@@ -0,0 +1,147 @@
+            vcpkg-triplet=${{ env.vcpkg-triplet }}
+
+      - name: Upgrade Dependencies
+        if: steps.cache-vcpkg.outputs.cache-hit == 'true'
+        run: |
+          vcpkg upgrade --no-dry-run --triplet ${{ env.vcpkg-triplet }}
+
+      - name: Build Dependencies
+        run: |
+          vcpkg install `@vcpkg.txt --triplet ${{ env.vcpkg-triplet }}
+          vcpkg list --x-full-desc
+
+      - name: Configure Platform Arch
+        run: |
+          if ("${{ matrix.arch }}" -eq "x86") {
+            $cmake_arch = "Win32"
+            $qt_arch = "win32_msvc2017"
+          } else {
+            $cmake_arch = "${{ matrix.arch }}"
+            $qt_arch = "win64_msvc2017_64"
+          }
+          Write-Host "::set-env name=CMAKE_GENERATOR_PLATFORM::$cmake_arch"
+          Write-Host "::set-env name=QT_ARCH::$qt_arch"
+
+      # Qt5 takes roughly a thousand years to build, so we download it from elsehwere...
+      - name: Restore Qt Cache
+        id: cache-qt
+        uses: actions/cache@v1
+        with:
+          path: ${{ github.workspace }}\qt
+          key: os=${{ runner.os }} qt=${{ matrix.qt }} arch=${{ env.qt_arch }}
+
+      - name: Install Qt
+        uses: jurplel/install-qt-action@v2
+        with:
+          arch: ${{ env.QT_ARCH }}
+          version: ${{ matrix.qt }}
+          dir: ${{ github.workspace }}\qt
+          cached: ${{ steps.cache-qt.outputs.cache-hit }}
+          aqtversion: ==0.9.*
+
+      # Build KF5SyntaxHighlighting manually due to Qt dependency...
+      - name: Checkout KF5SyntaxHighlighting
+        uses: actions/checkout@v2
+        with:
+          repository: KDE/syntax-highlighting
+          ref: v5.66.0
+          path: syntax-highlighting
+
+      - uses:  shogo82148/actions-setup-perl@v1
+      - name: Build KF5SyntaxHighlighting
+        run: |
+          cd syntax-highlighting
+          mkdir build && cd build
+          cmake `
+            -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake `
+            -DVCPKG_TARGET_TRIPLET=${{ env.vcpkg-triplet }} `
+            -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}\install `
+            -DBUILD_HTML_DOCS=OFF -DBUILD_MAN_DOCS=OFF -DBUILD_QTHELP_DOCS=OFF -DBUILD_TESTING=OFF `
+            ..
+          cmake --build . --config Release -j 2
+          cmake --build . --config Release --target INSTALL
+
+      - name: Checkout HSPlasma
+        uses: actions/checkout@v2
+        with:
+          repository: H-uru/libhsplasma
+          path: libhsplasma
+
+      - name: Build libHSPlasma
+        run: |
+          cd libhsplasma
+          mkdir build && cd build
+          cmake `
+            -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake `
+            -DVCPKG_TARGET_TRIPLET=${{ env.vcpkg-triplet }} `
+            -DCMAKE_INSTALL_PREFIX=${{ github.workspace }}\install `
+            -DENABLE_PYTHON=OFF -DENABLE_TOOLS=OFF -DENABLE_NET=OFF -DENABLE_PHYSX=OFF ..
```

Comment:
Not required for this PR, but when we start building installers, we need to set ENABLE_PHYSX=ON for Win32 builds to match what the existing installers provide.